internal/dag/serviceapis_processor.go

```diff
@@ -26,5 +35,126 @@ func (p *ServiceAPIsProcessor) Run(dag *DAG, source *KubernetesCache) {
 	p.dag = dag
 	p.source = source
 
-	// NOT IMPLEMENTED
+	// reset the processor when we're done
+	defer func() {
+		p.dag = nil
+		p.source = nil
+	}()
+
+	for _, route := range p.validHTTPRoutes() {
+		p.computeHTTPRoute(route)
+	}
+}
+
+func (p *ServiceAPIsProcessor) computeHTTPRoute(route *serviceapis.HTTPRoute) {
+
+	// Determine the hosts on the route, if no hosts
+	// are defined, then set to "*".
+	var hosts []string
+	if len(route.Spec.Hostnames) == 0 {
+		hosts = append(hosts, "*")
+	} else {
+		for _, host := range route.Spec.Hostnames {
```

Comment:
I suggest validating each hostname. The first phase of hostname validation can be:
- Ensure the hostname is not an IP address. xref: https://golang.org/pkg/net/#ParseIP
- Ensure the hostname does not contain a port. xref: https://golang.org/pkg/net/#SplitHostPort

As a second phase, the `hostname` spec states "Hostname is the fully qualified domain name of a network host, as defined by RFC 3986."  We should be able to use https://golang.org/pkg/net/?m=all#isDomainName (verify that it supports wildcard prefixes or augment to support wildcards).

Reply:
#3353